### PR TITLE
python-telegram-bot: init at 9.0.0

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build python2Packages.python-telegram-bot python3Packages.python-telegram-bot

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/python-telegram-bot/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "python-telegram-bot";
+  version = "9.0.0";
+  name = "${pname}-${version}";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0a5b4wfc6ms7kblynw2h3ygpww98kyz5n8iibqbdyykwx8xj7hzm";
+  };
+
+  propagatedBuildInputs = [ certifi future ];
```
[the remark this answers is not in the document]
Comment (FRidh) [reply pkgs/development/python-modules/python-telegram-bot/default.nix]: If one wants those they can just add those packages to the closure in a Python env.

File: pkgs/development/python-modules/python-telegram-bot/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  pname = "python-telegram-bot";
+  version = "9.0.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0a5b4wfc6ms7kblynw2h3ygpww98kyz5n8iibqbdyykwx8xj7hzm";
+  };
+
+  propagatedBuildInputs = [ certifi future ];
+  doCheck = !isPy3k;
```
Comment (FRidh) [inline pkgs/development/python-modules/python-telegram-bot/default.nix]: What is the reason for this? Can you include a comment.

Comment (dotlambda) [reply pkgs/development/python-modules/python-telegram-bot/default.nix]: The reason is that 0 tests are run when using Python 2, so it should really be just `doCheck = false` for both.
But I made one test run by adding `pysocks` as a dependency. I believe this is possible for the 7 remaining failing tests as well.

Comment (dotlambda) [reply pkgs/development/python-modules/python-telegram-bot/default.nix]: Turns out the tests need network access (obviously). So it should be `doCheck = false`.

**Comment by dotlambda** [comment]
The package uses a vendored version of `urllib3`. Just as [ArchLinux does](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=python-telegram-bot), I propose the following change:
```
diff --git a/pkgs/development/python-modules/python-telegram-bot/default.nix b/pkgs/development/python-modules/python-telegram-bot/default.nix
index b753e8db70f..d5ce74ff942 100644
--- a/pkgs/development/python-modules/python-telegram-bot/default.nix
+++ b/pkgs/development/python-modules/python-telegram-bot/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchPypi, buildPythonPackage, isPy3k, certifi, future }:
+{ stdenv, fetchPypi, buildPythonPackage, certifi, future, urllib3 }:
 
 buildPythonPackage rec {
   pname = "python-telegram-bot";
@@ -10,8 +10,16 @@ buildPythonPackage rec {
     sha256 = "0a5b4wfc6ms7kblynw2h3ygpww98kyz5n8iibqbdyykwx8xj7hzm";
   };
 
-  propagatedBuildInputs = [ certifi future ];
-  doCheck = !isPy3k;
+  prePatch = ''
+    rm -rf telegram/vendor
+    substituteInPlace telegram/utils/request.py \
+      --replace "import telegram.vendor.ptb_urllib3.urllib3 as urllib3" "import urllib3 as urllib3" \
+      --replace "import telegram.vendor.ptb_urllib3.urllib3.contrib.appengine as appengine" "import urllib3.contrib.appengine as appengine" \
+      --replace "from telegram.vendor.ptb_urllib3.urllib3.connection import HTTPConnection" "from urllib3.connection import HTTPConnection" \
+      --replace "from telegram.vendor.ptb_urllib3.urllib3.util.timeout import Timeout" "from urllib3.util.timeout import Timeout"
+  '';
+
+  propagatedBuildInputs = [ certifi future urllib3 ];
 
   meta = with stdenv.lib; {
     description = "This library provides a pure Python interface for the Telegram Bot API.";
```

**Review by dotlambda — APPROVED** [review]

**Pull request merged**